src/Symfony/Bundle/FrameworkBundle/Command/CachePoolDeleteCommand.php

```
@@ -0,0 +1,79 @@
+
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\HttpKernel\CacheClearer\Psr6CacheClearer;
+
+/**
+ * Delete an item from a cachhe pool.
+ *
+ * @author Pierre du Plessis <pdples@gmail.com>
+ */
+final class CachePoolDeleteCommand extends Command
+{
+    protected static $defaultName = 'cache:pool:delete';
+
+    private $poolClearer;
+
+    public function __construct(Psr6CacheClearer $poolClearer)
+    {
+        parent::__construct();
+
+        $this->poolClearer = $poolClearer;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setDefinition(array(
+                new InputArgument('pool', InputArgument::REQUIRED, 'The cache pool to delete an item from'),
+                new InputArgument('key', InputArgument::REQUIRED, 'The cache key to delete from the pool'),
+            ))
+            ->setDescription('Deletes an item from a cache pool')
+            ->setHelp(<<<'EOF'
+The <info>%command.name%</info> deletes an item from a given cache pool.
+
+    %command.full_name% <pool> <key>
+EOF
+            )
+        ;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $io = new SymfonyStyle($input, $output);
+        $pool = $input->getArgument('pool');
+        $key = $input->getArgument('key');
+        $cachePool = $this->poolClearer->getPool($pool);
+
+        if (!$cachePool->hasItem($key)) {
+            return $io->warning(sprintf('Cache item "%s" does not exist in cache pool "%s"', $key, $pool));
+        }
+
+        if (!$cachePool->deleteItem($key)) {
+            return $io->error(sprintf('Cache item "%s" could not be deleted', $key));
```

The warning and error functions do not return a status code, however, I would imagine that _at least_ the error should return a status code other than 0.

---

I'm aware that they don't return a status code, but for me this is just cleaner than

```
$io->error(sprintf('Cache item "%s" could not be deleted', $key));

return;
```

---

For the `error` I think it might be better to rather throw an exception, which should then have a proper error code set

---

go for an exception then :)